spec/ddtrace/workers/runtime_metrics_spec.rb

```diff
@@ -266,15 +272,17 @@
 
   describe 'integration tests', :integration do
     describe 'interval' do
+      include_context 'short default flush interval'
+
       after { worker.stop }
 
-      it 'produces metrics every 10 seconds' do
+      it 'produces metrics every interval' do
         worker.perform
 
-        sleep 10.1
+        sleep(default_flush_interval + 0.1)
 
         # Metrics are produced once right away
-        # and again after 10 seconds.
+        # and again after an interval.
         expect(metrics).to have_received(:flush).twice
```

Comment:
Hmmm this looks like it has the potential to become a flaky test. Perhaps this should be turned into an `wait_for`?

Reply:
`try_wait_until` only works on a condition; I don't think we can use a RSpec mock expectation as a condition, as it raises an error.

Reply:
Hmm yeah. I guess this needs a bit of a rewrite if it turns out to be flaky. I always suspect tests with `sleep` 🤔